File: Cryptographic Frontend/src/methods/SignMethod.cpp
#include <methods/SignMethod.hpp>

#include <StringArgument.hpp>
#include <ResponseMessage.hpp>

#include <json/json.h>

using namespace cf;

SignMethod::SignMethod(std::string data)
    : Method("Sign")
{
    addArgument(ArgumentPtr(new StringArgument("data", data)));
}

ResponseMessageStrategy SignMethod::getStrategy() const
{
      return [](std::string const& message) -> ResponseMessagePtr { 
      Json::Value json;
      Json::Reader reader;
      
      reader.parse(message, json);
      ResponseMessagePtr response(new ResponseMessage());
      response->setValue("signedData", json["signedData"].asString());
      
      return std::move(response);
  };
}

File: Cryptographic_Backend/src/main/java/cl/niclabs/cb/backend/methods/SignMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;
import javax.xml.bind.DatatypeConverter;

public interface SignMethod extends Method {

  public static class ReturnValue {
      public final String signedData;

      public ReturnValue(String signedData) {
          this.signedData = signedData;
      }
  }

  public static class Args {
    public final String data;

      public Args(String data) {
          this.data = data;
      }
  }

}
